File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {

    unordered_map< int , int > check ;
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        for(int i = 0 ; i < nums.size() ; i++)
        {
            int diff = target - nums[i];

            if(check.find(diff) != check.end())
            {
                int val = check[diff];
                // int arr = 
                return {val , i};;
            }
            
                check[nums[i]]=i;
            
        }
        return {};
    }
};

File: Data Structures & Algorithms/search-insert-position/submission-0.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        
        return bs(nums , target);
    }

    public int bs(int[] n , int t)
    {
        int l = 0 , r = n.length - 1 ;
        int m ;
        while(l<=r)
        {
            m = l +(r-l)/2;

            if(t == n[m])return m;
            else if(t<n[m]) r= m -1;
            else l = m+1;
        }

        return l;
    }
}